src/App.tsx

```diff
@@ -15,71 +15,84 @@ export const goods = [
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  addSelectedGood = (good: string) => {
+    this.setState({ selectedGood: good });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  removeSelectedGood = () => {
+    this.setState({ selectedGood: '' });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood ? `${selectedGood} is selected` : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+          {selectedGood && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.removeSelectedGood}
+            />
+          )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(good => {
+              const isSelectedGood = selectedGood === good;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  className={
+                    isSelectedGood
+                      ? 'has-background-success-light'
+                      : ''
+                  }
+                >
+                  <td>
+                    <button
+                      data-cy={isSelectedGood
+                        ? 'RemoveButton'
+                        : 'AddButton'}
+                      type="button"
+                      className={isSelectedGood
+                        ? 'button is-info'
+                        : 'button'}
+                      onClick={
+                        isSelectedGood
+                          ? this.removeSelectedGood
+                          : () => this.addSelectedGood(good)
+                      }
+                    >
+                      {isSelectedGood ? '+' : '-'}
+                    </button>
```

Comment:
```suggestion
        {isSelectedGood ? (
                    <button
                      data-cy='RemoveButton'
                      type="button"
                      className='button is-info'
                      onClick={this.removeSelectedGood}
                    >
                      +
                    </button> ) : (...)
```
since we are changing almost all the props, it is better to write it through the ternary operator, it will be much more readable